tests/smartrowheightcache/smartrowheightcachetest.cpp

```diff
@@ -0,0 +1,326 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        tests/smartrowheightcache/smartrowheightcachetest.cpp
+// Purpose:     smart row height cache for dataview unit test
+// Author:      Jens Goefpert
+// Created:     2018-03-06
+// Copyright:   (c) 2018 Jens Goepfert
+///////////////////////////////////////////////////////////////////////////////
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+#include "testprec.h"
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#ifndef WX_PRECOMP
+#endif
+
+#include "wx/private/smartrowheightcache.h"
+
+// ----------------------------------------------------------------------------
+// local functions
+// ----------------------------------------------------------------------------
+
+
+// ----------------------------------------------------------------------------
+// test class
+// ----------------------------------------------------------------------------
+
+class SmartRowHeightCacheTestCase : public CppUnit::TestCase
```

Comment:
I won't ask you to change this if you don't want to, but it would be great if new tests could be written without the compatibility CppUnit macros and just use CATCH (which is used now under the hood even for such tests anyhow) directly.

This would make things simpler as you can just write `TEST_CASE() { ... your test code here ... }` without bothering with defining any classes, suites etc and more convenient to use during run-time as you would be able to select just a particular test to run.

For an example of a test using CATCH directly, see `tests/html/htmprint.cpp` for example.

Reply:
it is a CATCH test now